testing/kuttl/e2e/exporter-custom-queries/00-assert.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+commands:
+# First, check that all containers in the instance pod are ready
+# Then, list the query files mounted to the exporter and check for expected files
+# Then, check the contents of the queries to ensure queries.yml was generated correctly
+# Finally, store the current exporter pid as an annotation
+- script: |
+    retry() { bash -ceu 'printf "$1\nSleeping...\n" && sleep 5' - "$@"; }
```

Comment:
I almost want to move the `exit 1` into `retry` so then we can say something like, "when this func is called, we sleep, then fail, to signal to KUTTL to restart this test step"

Reply:
i like that

Reply:
I'm not sure if this would work :thinking: would that just exit the `retry` function?

Reply:
hmmm, what if we `set -e` at the top of the script? Not sure abt that, just a thought to maybe try. 

I'm curious: if you take `retry` out and just put in a sleep, does anything get printed if the script fails?

Reply:
I did a quick test and `set -e` does return as expected.

> I'm curious: if you take retry out and just put in a sleep, does anything get printed if the script fails?

I'm not tracking... like this?

```
{ contains x y } || { sleep 5 && exit 1}
```

Reply:
with that, there is no error message - i'm not sure what would happen if the error continued until the timeout 

the first command just sleeps and exits but the second uses retry
```
    logger.go:42: 09:58:34 | exporter-no-tls/0-00--create-cluster | running command: [sh -c set -e
        retry() { bash -ceu 'printf "$1\nSleeping...\n" && sleep 5 && exit 1' - "$@"; }
...
        pod=$(kubectl get pods -o name -n "${NAMESPACE}" \
          -l postgres-operator.crunchydata.com/cluster=exporter-no-tls \
          -l postgres-operator.crunchydata.com/crunchy-postgres-exporter=true)
        [ "$pod" = "" ] && sleep 5 && exit 1
...
    logger.go:42: 09:58:40 | exporter-no-tls/0-00--create-cluster | running command: [sh -c set -e
...
        condition_json=$(kubectl get "${pod}" -n "${NAMESPACE}" -o jsonpath="{.status.conditions}")
        [ "$condition_json" = "" ] && retry "conditions not found"
        { check_containers_ready "$condition_json"; } || {
          retry "containers not ready"
        }
...
    logger.go:42: 09:58:40 | exporter-no-tls/0-00--create-cluster | false
    logger.go:42: 09:58:40 | exporter-no-tls/0-00--create-cluster | containers not ready
    logger.go:42: 09:58:40 | exporter-no-tls/0-00--create-cluster | Sleeping...
    logger.go:42: 09:58:46 | exporter-no-tls/0-00--create-cluster | running command: [sh -c set -e

Reply:
Yeah, this 
```
{ contains x y } || { sleep 5 && exit 1}
```
is what I was thinking. I was wondering if KUTTL would report something when it gets to that `exit`, but no, thinking about it now, I don't see why it would.

That said, do you know where the `false` is coming from in 
```
logger.go:42: 09:58:40 | exporter-no-tls/0-00--create-cluster | false
```

Reply:
`check_containers_ready`... it takes the argument, the `.status.conditions` of the pod, uses `jq` and whatnot to get the `ContainersReady` status, and compares it to `True` and returns the result which in this case is `false`

Reply:
locally i have sent the output of `jq > /dev/null` so it doesn't have that line. I'll push that up shortly

Reply:
I didn't end up pushing this change. Will add to backlog